src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+    xfer->xferUnsignedInt(&m_flags);
+    xfer->xferReal(&m_preferredHeight);
+    xfer->xferReal(&m_preferredHeightDamping);
+    xfer->xferReal(&m_wanderAngle);
+    xfer->xferReal(&m_wanderLength);
+}
+
+float Locomotor::Get_Max_Speed_For_Condition(BodyDamageType condition) const
+{
+    float speed;
+
+    if (condition < g_theWriteableGlobalData->m_movementPenaltyDamageState) {
+        speed = m_template->m_maxSpeed;
+    } else {
+        speed = m_template->m_maxSpeedDamaged;
+    }
+
+    if (speed > m_maxSpeed) {
+        return m_maxSpeed;
+    }
+
+    return speed;
+}
+
+float Locomotor::Get_Max_Turn_Rate(BodyDamageType condition) const
+{
+    float rate;
+
+    if (condition < g_theWriteableGlobalData->m_movementPenaltyDamageState) {
+        rate = m_template->m_maxTurnRate;
+    } else {
+        rate = m_template->m_maxTurnRateDamaged;
+    }
+
+    if (rate > m_maxTurnRate) {
+        rate = m_maxTurnRate;
+    }
+
+    if (Get_Flag(ULTRA_ACCURATE)) {
+        return rate * 2.0f;
+    }
+
+    return rate;
+}
+
+float Locomotor::Get_Max_Acceleration(BodyDamageType condition) const
+{
+    float accel;
+
+    if (condition < g_theWriteableGlobalData->m_movementPenaltyDamageState) {
+        accel = m_template->m_acceleration;
+    } else {
+        accel = m_template->m_accelerationDamaged;
+    }
+
+    if (accel > m_maxAccel) {
+        return m_maxAccel;
+    }
+
+    return accel;
+}
+
+float Locomotor::Get_Max_Lift(BodyDamageType condition) const
+{
+    float lift;
+
+    if (condition < g_theWriteableGlobalData->m_movementPenaltyDamageState) {
+        lift = m_template->m_lift;
+    } else {
+        lift = m_template->m_liftDamaged;
+    }
+
+    if (lift > m_maxLift) {
+        return m_maxLift;
+    }
+
+    return lift;
+}
+
+float Locomotor::Get_Surface_Ht_At_Pt(float x, float y)
+{
+    float f = 0.0f;
+    float waterz;
+    float groundz;
+
+    if (g_theTerrainLogic->Is_Underwater(x, y, &waterz, &groundz)) {
+        return f + waterz;
+    } else {
+        return f + groundz;
+    }
+}
+
+float Locomotor::Get_Braking() const
+{
+    float braking = m_template->m_braking;
+
+    if (braking > m_maxBreaking) {
+        return m_maxBreaking;
+    }
+
+    return braking;
+}
+
+void Locomotor::Loco_Update_Move_Towards_Position(
+    Object *obj, const Coord3D &goal_pos, float on_path_dist_to_goal, float desired_speed, bool *blocked)
+{
+    Set_Flag(MAINTAIN_POS_IS_VALID, false);
+    float maxspeed = Get_Max_Speed_For_Condition(obj->Get_Body_Module()->Get_Damage_State());
+
+    if (desired_speed > maxspeed) {
+        desired_speed = maxspeed;
+    }
+
+    float speed = maxspeed / Get_Braking() * maxspeed / 2.0f;
+
+    if (on_path_dist_to_goal > 10.0f && on_path_dist_to_goal > speed) {
+        Set_Flag(IS_BRAKING, false);
+        m_brakingFactor = 1.0f;
+    }
+
+    PhysicsBehavior *physics = obj->Get_Physics();
+
+    if (physics != nullptr) {
+        if ((m_template->m_surfaces & 8) != 0
```

Comment:
Does flag 8 have a name possibly?

Does surfaces correspond to

```
enum Surfaces
{
    SURFACE_GROUND,
    SURFACE_AIR,
    SURFACE_GROUND_AIR,
    SURFACE_COUNT,
};
```
?

If yes, is it missing enum values?

Reply:
It's not that, haven't been able to work out what the flags for this are yet.